docs/src/test/scala/sample/scaladsl/ConsumerExample.scala

```diff
@@ -341,3 +342,51 @@ object ExternallyControlledKafkaConsumer extends ConsumerExample {
   }
 }
 
+class StreamWrapperActor extends Actor with ConsumerExample with ActorLogging {
+
+  def receive = {
+    case _ =>
+  }
+
+  def processMsg(g: ConsumerRecord[Array[Byte], String]): Future[String] = Future.successful("")
+
+  def createStream(): Unit = {
+    //#errorHandlingStop
+    val done =
+      Consumer.plainSource(consumerSettings, Subscriptions.topics("topic1"))
+        .mapAsync(1)(processMsg)
+        .runWith(Sink.ignore)
+
+    done.onComplete {
+      case Failure(ex) =>
+        log.error(ex, "Stream failed, stopping the actor.")
+        context.stop(self)
```

Comment:
Thanks, I was explaining the same to someone and decided to write this docs, then copied the bad code here :)